### PR TITLE
Api changes

### PR DESCRIPTION
**Pull request by tfwright**
implementing some new methods, updating old ones

### CODE REVIEW

File: spec/squall/base_spec.rb [hunk 1]
```diff
@@ -9,17 +9,16 @@ class CamelTest < Squall::Base;end;
...
 describe Squall::Base do
   before(:each) do
-    default_config
     @base = Squall::Base.new
   end
 
   describe "#initialize" do
     it "sets the base_uri" do
-      @base.class.base_uri.should == 'http://www.example.com'
+      @base.class.base_uri.should_not == nil
```
Comment (jmazzi) [inline spec/squall/base_spec.rb:17]:
``` ruby
should_not be_nil
```

File: spec/squall/virtual_machine_spec.rb [hunk 2]
```diff
@@ -283,18 +300,39 @@
...
+
+    it "deletes a virtual_machine" do
+      @virtual_machine.set_vip(1)
+      @virtual_machine.success.should be_true
+    end
+    
+    it "sets the vip status to false if currently true" do
+      pending "No way to actually test this without being able to interact with server state" do
+        result = @virtual_machine.set_vip(1)
+        result['virtual_machine']['vip'].should == "false"
```
Comment (jmazzi) [inline spec/squall/virtual_machine_spec.rb]: Are they returning a string here?

File: spec/vcr_cassettes/virtual_machine/change_owner.yml [hunk 3]
```diff
@@ -39,13 +39,13 @@
...
   request: !ruby/struct:VCR::Request 
     method: :post
-    uri: http://user:pass@www.example.com:80/virtual_machines/1/change_owner.json?user_id=404
+    uri: http://<USER>:<PASS>@<URL>:80/virtual_machines/1/change_owner.json?user_id=404
     body: 
     headers: 
       content-type: 
       - application/json
       authorization: 
-      - Basic WHAT
```
Comment (jmazzi) [inline spec/vcr_cassettes/virtual_machine/change_owner.yml:48]: WHAT!? YEAAAAA

Comment (tfwright) [inline spec/vcr_cassettes/virtual_machine/change_owner.yml:48]: WHAT??

`<REDACTED>`

File: spec/squall/virtual_machine_spec.rb (hunk 2 above)
Comment (tfwright) [inline spec/squall/virtual_machine_spec.rb]: no, but it should a pending test...i guess that's not the best way to make sure it flunks

**Pull request merged**